src/modifiers.rs

```diff
@@ -226,3 +230,116 @@ impl UpdateModifier for AccelModifier {
+    Cubic,
+
+    /// None is the default. No force field is applied.
+    None,
+}
+
+impl Default for ForceType {
+    fn default() -> Self {
+        ForceType::None
+    }
+}
+
+impl ForceType {
+    /// Converts the instance of ForceType into an integer, preparing to be sent to the GPU.
+    pub fn to_int(self) -> i32 {
+        match self {
+            ForceType::Linear => 1,
+            ForceType::Quadratic => 2,
+            ForceType::Cubic => 3,
+            ForceType::None => -1,
+        }
+    }
+}
+
+/// Parameters for the components making the force field.
+#[derive(Clone, Copy)]
+pub struct ForceFieldParam {
+    /// The particle_update.wgsl shader interprets this field as a position when the force type is set
+    /// to either [`ForceType::Linear`], [`ForceType::Quadratic`] or [`ForceType::Cubic`].
+    pub position: Vec3,
+    /// For a attracting/repulsing force, this is the maximum radius of the sphere of influence, outside of which
+    /// the force field is null.
+    pub max_radius: f32,
+    /// For a attracting/repulsing force, this is the minimum radius of the sphere of influence, inside of which
+    /// the force field is null, avoiding the singularity at the source position.
+    pub min_radius: f32,
+    /// The intensity of the force is proportional to mass. Note that the particles_update.wgsl shader will
+    /// ignore all subsequent force field components after it encounters a component with a mass of zero.
+    /// To change the force from an attracting one to a repulsive one, simply set the mass to a negative value.
+    pub mass: f32,
+    /// Defines whether the field is proportional to the distance from the source linearly, quadratically,
+    /// cubicly or not at all.
+    pub force_type: ForceType,
+    /// If set to true, the particles that enter within the `min_radius` will conform to a sphere around the
+    /// source position, appearing like a recharging effect.
+    pub conform_to_sphere: bool,
+}
+
+impl Default for ForceFieldParam {
+    fn default() -> Self {
+        // defaults to no force field (a mass of 0)
+        ForceFieldParam {
+            position: Vec3::new(0., 0., 0.),
+            min_radius: 0.1,
+            max_radius: 0.0,
+            mass: 0.,
+            force_type: ForceType::None,
+            conform_to_sphere: false,
+        }
+    }
+}
+
+/// A modifier to apply a force field to all particles each frame. The force field is made up of
+/// point sources, also called 'components'. The maximum number of components is set with [`FFNUM`].
+#[derive(Default, Clone, Copy)]
+pub struct ForceFieldModifier {
+    /// Array of force field components.
+    pub force_field: [ForceFieldParam; FFNUM],
+}
+
+impl ForceFieldModifier {
+    /// Instantiate a ForceFieldModifier.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the number of attractors exceeds [`FFNUM`].
+    pub fn new(point_attractors: Vec<ForceFieldParam>) -> Self {
```

Comment:
Maybe
```suggestion
    pub fn new(point_attractors: &[ForceFieldParam]) -> Self {
```
is more generic/useable? Ideally even `impl IntoIter<ForceFieldParam>` is the most generic way, since all we need here is an iterator.

Reply:
Good idea!